api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -1010,6 +1044,76 @@ static String notebookLockingEmailHash(String bucket, String email) {
     return ResponseEntity.ok(recentWorkspaceResponse);
   }
 
+  private WorkspaceResource createWorkspaceResourceFromWorkspace(DbWorkspace workspace) {
+    return new WorkspaceResource()
+        .workspaceFirecloudName(workspace.getFirecloudName())
+        .workspaceId(workspace.getWorkspaceId())
+        .workspaceNamespace(workspace.getWorkspaceNamespace())
+        .workspaceBillingStatus(workspace.getBillingStatus());
+  }
+
+  @Override
+  public ResponseEntity<WorkspaceResourceResponse> getDataPageItems(
```

Comment:
What about `getWorkspaceWithResources`? We try not to refer to UI constructs like individual pages/frames/panes from the backend.

Reply:
It doesn't return a workspace though. Maybe `getWorkspaceResources`? Except it doesn't return all resources. (There was an hour or so brainstorming in slack, but we couldn't really come up with something I liked. If you have other ideas, I would love to hear them)